### DISCUSSION TITLE
Diagnostic CLI command for i2c

### DISCUSSION THREAD

**Discussion by laurentopia** [iNavFlight/inav#5007] [Ideas]
## Current Behavior
I2C is a black box, a very very black box and a mag may not work for unknown reason

## Desired Behavior
some clarity, at least a list of i2c peripherals

## Suggested Solution
Add CLI command to list i2c peripheral addresses, it would help a lot

## Who does this impact? Who is this for?
bn880 users, some times auto mag doesn't work

[a comment by an automated account is not shown]

**Comment by digitalentity** [comment]
I think figuring out I2C device addresses is out of scope for INAV since we can't allow arbitrary addresses anyway to avoid odd behavior. Most of I2C issues are hardware-related (mis-wiring, lack of pull-up resistors etc).

**Comment by laurentopia** [comment]
This is not about allowing arbitrary addresses, it's about **listing** addresses of everything on the i2c bus to help diagnostic.
